core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -354,25 +404,29 @@ public TenantObject setAdapterConfigurations(final JsonArray configurations) {
      * Adds configuration information for a protocol adapter.
      * 
      * @param config The configuration properties to add.
-     * @throws NullPointerException if config is {@code null}.
-     * @throws IllegalArgumentException if the given configuration does not contain
-     *                a <em>type</em> name.
      * @return This tenant for command chaining.
-     * @throws NullPointerException if config is {@code null}.
+     * @throws NullPointerException if the given configuration is {@code null}.
+     * @throws IllegalArgumentException if the given configuration does not contain a <em>type</em> or
+     *                                  if any of the already existing adapters has the same <em>type</em> or
+     *                                  if error parsing the given configuration.
      */
     public TenantObject addAdapterConfiguration(final JsonObject config) {
 
         Objects.requireNonNull(config);
 
-        final Object type = config.getValue(TenantConstants.FIELD_ADAPTERS_TYPE);
-        if (String.class.isInstance(type)) {
-            if (adapterConfigurations == null) {
-                adapterConfigurations = new LinkedList<>();
-            }
-            adapterConfigurations.add(config.getMap());
+        final Adapter adapter = config.mapTo(Adapter.class);
+
+        if (adapters == null) {
+            adapters = new LinkedList<>();
         } else {
-            throw new IllegalArgumentException("adapter configuration must contain type field");
+            final boolean hasAdapterOfSameType = adapters.stream()
```

Comment:
isn't there a method for this already, e.g. *getAdapter(String)*?

Reply:
Yes, i have replaced it now with _getAdapter(String)_.